daemon/prune.go

```diff
@@ -150,3 +154,72 @@ func (daemon *Daemon) ImagesPrune(config *types.ImagesPruneConfig) (*types.Image
 
 	return rep, nil
 }
+
+// localNetworksPrune removes unused local networks
+func (daemon *Daemon) localNetworksPrune(config *types.NetworksPruneConfig) (*types.NetworksPruneReport, error) {
+	rep := &types.NetworksPruneReport{}
+	var err error
+	// When the function returns true, the walk will stop.
+	l := func(nw libnetwork.Network) bool {
+		nwName := nw.Name()
+		predefined := runconfig.IsPreDefinedNetwork(nwName)
+		if !predefined && len(nw.Endpoints()) == 0 {
```

Comment:
Thank you for looking into this.

However, `NewRequestForbiddenError` is not a type.
https://github.com/docker/docker/blob/c452e1bfe6b19d44bf2ed2a1eb8aef7e23d6bafb/api/errors/errors.go#L33

```go
func NewRequestForbiddenError(err error) error {
	return NewErrorWithStatusCode(err, http.StatusForbidden)
}

func NewErrorWithStatusCode(err error, code int) error {
	return apiError{err, code}
}

type apiError struct {
	error
	statusCode int
}

func (e apiError) HTTPErrorStatusCode() int {
	return e.statusCode
}
```

Since `NewReuqestForbiddenError()` returns a non-exposed `apiError`, it is hard to check the error in `daemon/prune.go`.
Of course  we can check the error by exposing `apiError`, or just creating an interface for `HTTPErrorStatusCode()`, or even checking the error string.
However, it would be too much for the purpose.

WDYT?